autoload/seak.vim

```diff
@@ -189,14 +225,86 @@ endif
 "
 if has('nvim')
   function! s:close(id) abort
-    call nvim_buf_del_extmark(0, s:ns, a:id)
+    call nvim_buf_del_extmark(s:get_current_bufnr(), s:ns, a:id)
   endfunction
 else
   function! s:close(id) abort
     call prop_remove({
     \   'type': 'seak',
     \   'id': a:id,
+    \   'bufnr': s:get_current_bufnr(),
     \ })
   endfunction
 endif
 
+"
+" s:getcmdline
+"
+" Similar to the built-in getcmdline(), but cooperate with cmdwin.
+"
+function! s:getcmdline() abort
+  if s:is_in_cmdwin()
+    return getline('.')
+  else
+    return getcmdline()
+  endif
+endfunction
+
+"
+" s:getcmdtype
+"
+" Similar to the built-in getcmdtype(), but cooperate with cmdwin.
+"
+function! s:getcmdtype() abort
+  let l:cmdtype = getcmdtype()
+  if l:cmdtype !=# ''
+    return l:cmdtype
+  else
+    return getcmdwintype()
+  endif
+endfunction
+
+"
+" s:is_in_cmdwin
+"
+function! s:is_in_cmdwin() abort
+  return getcmdwintype() !=# '' && getcmdtype() ==# ''
+endfunction
+
+"
+" s:get_current_bufnr
+"
+function! s:get_current_bufnr() abort
+  if s:is_in_cmdwin()
+    return bufnr('#')
+  endif
+  return bufnr('%')
+endfunction
+
+"
+" s:get_current_bufline
+"
+function! s:get_current_bufline(lnum, ...) abort
+  return call('getbufline', [s:get_current_bufnr(), a:lnum] + a:000)
+endfunction
+
+"
+" s:get_current_winID
+"
+function! s:get_current_winID() abort
+  return bufwinid(s:get_current_bufnr())
+endfunction
+
+"
+" s:getcurpos_on_current_window
+"
+if has('nvim')
+  function! s:getcurpos_on_current_window() abort
+    let l:winID = s:get_current_winID()
+    return [0, line('.', l:winID), col('.', l:winID), 0, 0]
```

Comment:
Unfortunately, the neovim doesn't accept winid on line/col too...

```
    let l:cursor = nvim_win_get_cursor(s:get_current_winID())
    return [s:get_current_bufnr(), l:cursor[0], l:cursor[1] + 1]
```

Reply:
From `:h getpos()` and `:h getcurpos()`:
```
getpos({expr})	Get the position for String {expr}.  For possible values of
		{expr} see |line()|.  For getting the cursor position see
		|getcurpos()|.
		The result is a |List| with four numbers:
		    [bufnum, lnum, col, off]
		"bufnum" is zero, unless a mark like '0 or 'A is used, then it
		is the buffer number of the mark.
```

```
getcurpos([{winid}])
		Get the position of the cursor.  This is like getpos('.'), but
		includes an extra "curswant" item in the list:
		    [0, lnum, col, off, curswant] ~
```


According to this doc, the `bufnum` will be 0 when `'.'` is specified as `{expr}`.
So I think `s:get_current_bufnr()` is unnecessary.

Reply:
```
    let l:cursor = nvim_win_get_cursor(s:get_current_winID())
    return [0, l:cursor[0], l:cursor[1] + 1]
```
It's OK.

Reply:
Fixed. Thank you.